### PR TITLE
Add logic to cache the RPATH variables in CachedCMakePackages.

### PR DESCRIPTION
**Pull request by bvanessen**

### CODE REVIEW

**Comment by bvanessen** [comment]
@spackbot fix style

File: lib/spack/spack/build_systems/cached_cmake.py [hunk 1]
```diff
@@ -285,13 +286,29 @@ def initconfig_hardware_entries(self):
     def std_initconfig_entries(self):
         cmake_prefix_path_env = os.environ["CMAKE_PREFIX_PATH"]
         cmake_prefix_path = cmake_prefix_path_env.replace(os.pathsep, ";")
+        cmake_rpaths_env = spack.build_environment.get_rpaths(self.pkg)
+        cmake_rpaths_path = ";".join(cmake_rpaths_env)
+        complete_rpath_list = cmake_rpaths_path
+        if "SPACK_COMPILER_EXTRA_RPATHS" in os.environ:
+            spack_extra_rpaths_env = os.environ["SPACK_COMPILER_EXTRA_RPATHS"]
+            spack_extra_rpaths_path = spack_extra_rpaths_env.replace(os.pathsep, ";")
+            complete_rpath_list = "{0};{1}".format(complete_rpath_list, spack_extra_rpaths_path)
```
Comment (becker33) [inline lib/spack/spack/build_systems/cached_cmake.py:295]: Aren't these supposed to be `+=`?

**Review by becker33 — CHANGES_REQUESTED** [review]

File: lib/spack/spack/build_systems/cached_cmake.py (hunk 1 above)
Comment (bvanessen) [reply lib/spack/spack/build_systems/cached_cmake.py:295]: @becker33 I think that since I have line 295, it will append.

**Review by becker33 — APPROVED** [review]

**Comment by adrienbernede** [comment]
@bvanessen could you explain why this was necessary ? I genuinely wonder.
Our CachedCMakePackages are working well without this, even when built outside of Spack using a Spack built stack of dependencies.

**Pull request merged**